lib/kafo/kafo_configure.rb

```diff
@@ -43,30 +43,45 @@ def initialize(*args)
       self.class.preset_color_scheme
       self.class.logger           = Logger.new
       self.class.exit_handler     = ExitHandler.new
-      self.class.scenario_manager = ScenarioManager.new((defined?(CONFIG_DIR) && CONFIG_DIR) || (defined?(CONFIG_FILE) && CONFIG_FILE))
+      @progress_bar               = nil
+      @config_reload_requested    = false
+
+      scenario_manager = setup_scenario_manager
+      self.class.scenario_manager = scenario_manager
 
       # Handle --list-scenarios before we need them
-      self.class.scenario_manager.list_available_scenarios if ARGV.include?('--list-scenarios')
+      scenario_manager.list_available_scenarios if ARGV.include?('--list-scenarios')
+      setup_config(config_file)
+
+      self.class.hooking.execute(:pre_migrations)
+      # run migrations
+
+      # reload config
+      if @config_reload_requested
+        scenario_manager = setup_scenario_manager
```

Comment:
why creating `scenario_manager` local var? it does not seem to be used later again, so `self.class.scenario_manager = setup_scenario_manager` should be fine